src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -124,20 +124,32 @@ bool TNumberSchemaItem<NumberType>::isValidNumberType(SmartType type) {
   }
 }
 
-template<typename NumberType>
-Errors::eType TNumberSchemaItem<NumberType>::validate(const SmartObject& Object) {
+template <typename InputType, typename OutputType>
+OutputType IntStaticCast(const InputType& value) {
```

Comment:
I think name like `SafeStaticCast` would be better for this function because it could be used not only with `int` type